src/Model/CodecCursor.php

```diff
@@ -61,22 +59,19 @@ public function current(): ?object
 
     /**
      * @template NativeClass of Object
-     * @param DocumentCodec<NativeClass> $codec
+     * @param CursorInterface<int, NativeClass> &Iterator<int, NativeClass> $cursor
+     * @param DocumentCodec<NativeClass>        $codec
      * @return self<NativeClass>
      */
-    public static function fromCursor(Cursor $cursor, DocumentCodec $codec): self
+    public static function fromCursor(CursorInterface&Iterator $cursor, DocumentCodec $codec): self
```

Comment:
Offhand, would Psalm actually complain about a redundant type hint once its stubs are updated for CursorInterface extending Iterator?

Reply:
Not sure - this went out the door with PHPC-2447 being implemented before I got done with these changes.